lib/UOSM-Sensor/Inc/Pressure_Sensor.hpp

```diff
@@ -0,0 +1,41 @@
+//
+// Created by Mohamad on 2023-09-05.
+//
+#include "Sensor.hpp"
+#include "ApplicationTypes.h"
+#include "MS5525DSO.h"
+#include "SPI.h"
+#include "Wire.h"
+#include <Arduino.h>
+
+
+
+class PressureSensor:public Sensor<pressure_t>{
+
+    public: 
+    
+    MS5525DSO sensor_pres = MS5525DSO(pp005GS); //Need to look into the input pp005GS
+
+    explicit PressureSensor(uint8_t buffer_size):Sensor<pressure_t>(buffer_size){
+        
+        Wire.begin();
+
+        if (!sensor_pres.begin()) {
+        Serial.println("failed to initialize pressure sensor!");
+        while (true) delay(100);
```

Comment:
This will cause the micro controller to freeze.
You can fix this by making the constructor throw an exception.

#include <exception>

throw std::runtime_error("Failed to initialize pressure sensor!")